lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs

```diff
@@ -387,6 +395,50 @@ testRule({
 				'the repeat() notation has non-trivial semantics and is currently not fixable (rows)',
 			message: messages.expected('grid-template'),
 		},
+		{
+			code: 'a { text-decoration-color: purple; text-decoration-thickness: 1px; text-decoration-style: solid; text-decoration-line: underline; }',
+			fixed: 'a { text-decoration: underline solid purple 1px; }',
+			message: messages.expected('text-decoration'),
+			description: 'CSS Text Decoration Module Level 4',
+		},
+	],
+});
+
+testRule({
+	ruleName,
+	config: [
+		true,
+		{
+			ignoreLonghands: [
+				'background-size',
+				'background-origin',
+				'background-clip',
+				'text-decoration-thickness',
+			],
+		},
+	],
+	fix: true,
+
```

Comment:
[suggest] Can you add `accept` cases, too?

Reply:
I couldn't think of good ones.
i.e. the accept ones needs to help with potential regressions

Do you have a suggestion?

Reply:
I see. Let's keep it.